PWGHF/TableProducer/HFDsToKKPiCandidateSelector.cxx

```diff
@@ -0,0 +1,154 @@
+  template <typename T1, typename T2>
+  bool selection(const T1& candidate, const T2& trackPion1, const T2& trackKaon, const T2& trackPion2)
+  {
+    auto candpT = candidate.pt();
+    int pTBin = findBin(pTBins, candpT);
+    if (pTBin == -1) {
+      return false;
+    }
+    // check that the candidate pT is within the analysis range
+    if (candpT < pTCandMin || candpT > pTCandMax) {
+      return false;
+    }
+    // cut on daughter pT
+    if (trackPion1.pt() < cuts->get(pTBin, "pT Pi") || trackKaon.pt() < cuts->get(pTBin, "pT K") || trackPion2.pt() < cuts->get(pTBin, "pT Pi")) {
+      return false;
+    }
+    // invariant-mass cut
+    if (std::abs(InvMassDsKKPi(candidate) - RecoDecay::getMassPDG(pdg::Code::kDs)) > cuts->get(pTBin, "deltaM")) {
+      return false;
+    }
+    if (std::abs(InvMassDsPiKK(candidate) - RecoDecay::getMassPDG(pdg::Code::kDs)) > cuts->get(pTBin, "deltaM")) {
+      return false;
+    }
+    // decay length cut
+    if (candidate.decayLength() < cuts->get(pTBin, "decay length")) {
+      return false;
+    }
+    if (candidate.decayLengthXYNormalised() < cuts->get(pTBin, "normalized decay length XY")) {
+      return false;
+    }
+    // cos. pointing angle cut
+    if (candidate.cpa() < cuts->get(pTBin, "cos pointing angle")) {
+      return false;
+    }
+    if (candidate.cpaXY() < cuts->get(pTBin, "cos pointing angle XY")) {
+      return false;
+    }
+    if (std::abs(candidate.maxNormalisedDeltaIP()) > cuts->get(pTBin, "max normalized deltaIP")) {
+      return false;
+    }
+    return true;
+  }
+
+  void process(aod::HfCandProng3 const& candidates, aod::BigTracksPID const&)
+  {
+    TrackSelectorPID selectorPion(kPiPlus);
+    selectorPion.setRangePtTPC(pidTPCMinpT, pidTPCMaxpT);
+    selectorPion.setRangeNSigmaTPC(-nSigmaTPC, nSigmaTPC);
+    selectorPion.setRangePtTOF(pidTOFMinpT, pidTOFMaxpT);
+    selectorPion.setRangeNSigmaTOF(-nSigmaTOF, nSigmaTOF);
+
+    TrackSelectorPID selectorKaon(selectorPion);
+    selectorKaon.setPDG(kKPlus);
+
+    // looping over 3-prong candidates
+    for (auto& candidate : candidates) {
+
+      // final selection flag:
+      auto statusDsToKKPi = 0;
+
+      if (!(candidate.hfflag() & 1 << DecayType::DsToKKPi)) {
+        hfSelDsToKKPiCandidate(statusDsToKKPi);
+        continue;
+      }
+      SETBIT(statusDsToKKPi, aod::SelectionStep::RecoSkims);
+
+      auto trackPos1 = candidate.index0_as<aod::BigTracksPID>(); // positive daughter (negative for the antiparticles)
+      auto trackNeg = candidate.index1_as<aod::BigTracksPID>();  // negative daughter (positive for the antiparticles)
+      auto trackPos2 = candidate.index2_as<aod::BigTracksPID>(); // positive daughter (negative for the antiparticles)
+
+      // topological selection
+      if ((!selection(candidate, trackPos1, trackNeg, trackPos2)) || !selection(candidate, trackPos2, trackNeg, trackPos1)) {
+        hfSelDsToKKPiCandidate(statusDsToKKPi);
+        continue;
```

Comment:
Why do you reject the candidate if it doesn't satisfy the selection with both track permutations?